cmd/bootstrap-post/main.go

```diff
@@ -1,80 +1,147 @@
 package main
 
 import (
-	"crypto/tls"
-	"fmt"
-	"net/http"
 	"os"
-	"tyk/tyk/bootstrap/data"
-	"tyk/tyk/bootstrap/helpers"
-	"tyk/tyk/bootstrap/readiness"
+	"strings"
+	"tyk/tyk/bootstrap/k8s"
+	"tyk/tyk/bootstrap/pkg/config"
+	"tyk/tyk/bootstrap/tyk"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	secretTypeOperator = "Operator"
+	secretTypePortal   = "Portal"
 )
 
 func main() {
-	err := data.InitPostInstall()
+	log := logrus.New()
+
+	conf, err := config.NewConfig()
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
 	}
 
-	err = readiness.CheckIfRequiredDeploymentsAreReady()
+	level, err := logrus.ParseLevel(conf.Log)
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
-	}
+		log.Infof(
+			"Failed to parse log level %v, continuing with the default log level Info, err %v", conf.Log, err,
+		)
 
-	tp := &http.Transport{
-		TLSClientConfig: &tls.Config{InsecureSkipVerify: data.BootstrapConf.InsecureSkipVerify},
+		level = logrus.InfoLevel
 	}
-	client := http.Client{Transport: tp}
 
-	fmt.Println("Started creating dashboard org")
+	log.SetLevel(level)
+	log.WithField("level", level.String()).Info("Set the log level")
 
-	err = helpers.CheckForExistingOrganisation(client)
+	k8sClient, err := k8s.NewClient(conf, log.WithField("Client", "Kubernetes"))
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
+	}
+
+	if err = k8sClient.CheckIfRequiredDeploymentsAreReady(); err != nil {
+		exit(log, err)
 	}
 
-	fmt.Println("Finished creating dashboard org")
-	fmt.Println("Generating dashboard credentials")
+	if conf.BootstrapDashboard {
+		conf.K8s.DashboardSvcUrl, err = k8sClient.DiscoverDashboardSvc()
+		if err != nil {
+			exit(log, err)
+		}
+	}
 
-	err = helpers.GenerateDashboardCredentials(client)
+	tykSvc := tyk.NewClient(conf, log.WithField("Client", "Tyk"))
+
+	orgExists, err := tykSvc.OrgExists()
 	if err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+		exit(log, err)
 	}
 
-	fmt.Println("Finished generating dashboard credentials")
-	fmt.Println("Started bootstrapping operator secret")
+	if !orgExists {
+		if conf.BootstrapDashboard {
+			log.Info("Bootstrapping Tyk Dashboard")
 
-	if data.BootstrapConf.OperatorKubernetesSecretName != "" {
-		err = helpers.BootstrapTykOperatorSecret()
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
+			if err = tykSvc.CreateOrganisation(); err != nil {
+				exit(log, err)
+			}
+
+			if err = tykSvc.CreateAdmin(); err != nil {
+				exit(log, err)
+			}
+		}
+
+		if conf.BootstrapPortal {
+			log.Info("Bootstrapping Tyk Classic Portal")
+
+			if err = tykSvc.BootstrapClassicPortal(); err != nil {
+				exit(log, err)
+			}
+
+			if err = k8sClient.RestartDashboard(); err != nil {
+				exit(log, err)
+			}
 		}
 	}
 
-	fmt.Println("Finished bootstrapping operator secret\nStarted bootstrapping portal secret")
+	// Common log message for organization existence
+	if orgExists && conf.DevPortalKubernetesSecretName != "" || conf.OperatorKubernetesSecretName != "" {
```

Comment:
Probably you need to change the condition as below
```suggestion
	if orgExists && (conf.DevPortalKubernetesSecretName != "" || conf.OperatorKubernetesSecretName) != "" {
```